usr/lib/common/lock_btree.c

```diff
@@ -502,7 +502,18 @@ void bt_init(struct btree *t, void (*delete_func)(void *))
      * Need a recursive mutex, because btree callback functions may call
      * btree functions again
      */
-    pthread_mutexattr_init(&attr);
-    pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE);
-    pthread_mutex_init(&t->mutex, &attr);
+    if (pthread_mutexattr_init(&attr) != 0) {
+        TRACE_ERROR("Initializing mutex attribute failed.\n");
+        return CKR_CANT_LOCK;
+    }
+    if (pthread_mutexattr_settype(&attr, PTHREAD_MUTEX_RECURSIVE) != 0) {
+        TRACE_ERROR("Initializing mutex attribute failed.\n");
```

Comment:
Although the trace will show the source code line numbers, we could use different trace messages for these 3 possible errors. Like 
TRACE_ERROR("pthread_mutexattr_init failed.\n");
TRACE_ERROR("pthread_mutexattr_settype failed.\n");
TRACE_ERROR("pthread_mutex_init failed.\n");

Reply:
sure, it was actually my intention to use 3 different messages, but somehow I forgot to change them.